pkg/apis/extensions/types.go

```diff
@@ -809,3 +810,75 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// +genclient=true
+
+// Template contains one or more partially complete configs with substitution parameters.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	Spec TemplateSpec `json:"spec"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Required: A yaml config list of resources to create
+	Objects []runtime.RawExtension `json:"objects"`
+}
```

Comment:
Non-string fields will be supported.  I need to better understand why we need $((NAME))  vs $(NAME) to support non-string fields.  Since we are unmarshaling the json into golang structs on the server side, is it sufficient to use the type of the field we are unmarshaling into to determine how we parse the field?  Given Parameter "TRUE" substituted with "true" can we say for string fields we will parse and assign the `string "true"` and for boolean fields we will parse and assign the `bool true`?